### DISCUSSION TITLE
Best Practice Guide

### DISCUSSION THREAD

**Discussion by vinc32** [Ysurac/openmptcprouter#3323] [Ideas]
Hi,

ist there any chance in providing something like guide for who should use which settings in which envirorment?
I'm already struggling a few days to get a good working solution.

Settings are not valid, just examples and not sure if they'll work ;) 

Example:

Situation One:
Hardware:
Highend (N100/N305/xxx)
Internet:
Fast 500mbps+ 
Settings: V2Ray Less, BBR2, OpenVPN TCP

Special Needs: if you need OMR Bypass - only use Shadowsocks / not working with V2Ray

Situaiton Two
Hardware:
Raspberry
Internet: <500mbps+
Settings: Shadowsocks, Cubic, MLVPN

Situation Three
....

**Comment by Ysurac** [comment]
Not really possible, there is too many possibilities.
For OMR-ByPass, this will be fixed in next release.

I made a script that test speed with all proxy/VPN, maybe, after some cleanup/work, I may release it and this may help.

**Reply by vinc32** [reply]
Thx for feedback :)

**Reply by github59173** [reply]
> Not really possible, there is too many possibilities. For OMR-ByPass, this will be fixed in next release.
> 
> I made a script that test speed with all proxy/VPN, maybe, after some cleanup/work, I may release it and this may help.

Sorry if I didn't see a future discussion about the script you are talking about but did that ever get released? Just curious what the best possible speed for my setup would be with best options used. (Using a proxmox VM, 64bit, i7 7700 with 6GB of RAM, complete overkill for a 2.5Gbps main line and a 5G hotspot). Sad I can't find a residential vps provider on the west coast of the USA for above 1Gbps for a reasonable price.

**Reply by Ysurac** [reply]
```omr-iperf-proxy``` is available in latest release to test TCP (upload and download with '-R' as parameter) or ```omr-test-speed``` to test via a simple http request.
Don't forget the VPS part that need to have a higher bandwidth.
Really no idea what speed can be reached with the router.
